packages/@aws-cdk/aws-ec2/lib/vpc-ref.ts

```diff
@@ -8,13 +44,13 @@ import { Construct, IDependable, Output, StringListOutput, Token } from "@aws-cd
  */
 export interface VpcPlacementStrategy {
     /**
-     * Whether to use the VPC's public subnets to start instances
+     * What subnet type to place the instances in
      *
-     * If false, the instances are started in the private subnets.
+     * By default, the instances are placed in the private subnets.
      *
-     * @default false
+     * @default SubnetType.Private
      */
-    usePublicSubnets?: boolean;
+    subnetsToUse?: SubnetType;
```

Comment:
Our VPCs commonly have an egress `PublicSubnet` and ingress `PublicSubnet` (per AZ). The separation is so we can place NACL on the egress and get some defense in depth plus avoid requiring an outbound layer 7 proxy for all situations. In order to use this we somehow need to place our bastion ASG in the ingress `PublicSubnet`, but right now we have no way to do that. One option is to allow an optional field called `subnetIds` as an escape back to what we know from traditional CloudFormation. Another option would be to enable some sort of selection based on tags (which isn't ready yet via #538).

Any opposition to adding a `subnets: VpcSubnetRef[]` or `subnetIds: string[]`. I think the former is more safe.

Reply:
I feel I would prefer tags or something similar. It doesn't have to be done all at once, we can add support for tag selection after that PR has landed.

In the mean time, we can also allow selecting by `name`, since we tagged them with names in the config.

Reply:
+1 on names or anything that gives us the ability to select to specific subnet when type is not distinct